stellarobservatory/quorums.py

```diff
@@ -0,0 +1,34 @@
+"""Torstens's quorum enumeration"""
+from .quorum_lachowski import greatest_quorum, next_split_node
+from .quorum_slices import get_dependencies_by_node
+
+
+def enumerate_quorums(slices_by_node):
+    """Enumerate all quorums of FBAS F (given by slices_by_node)."""
+    deps_by_node = get_dependencies_by_node(slices_by_node)
+    fbas_info = {
+        'deps_by_node': deps_by_node,
+        'slices_by_node': slices_by_node
+    }
+    all_nodes = set(slices_by_node.keys())  # all nodes need to be present as keys here
+    return traverse_quorums(fbas_info, set(), all_nodes)
+
+
+def traverse_quorums(fbas_info, committed, remaining):
+    """Given a FBAS F (by fbas_info) with set of nodes V
+    and given the sets: committed ⊆ V; R ⊆ V\\committed,
+    enumerate all quorums Q of F with committed ⊆ Q ⊆ committed ∪ remaining"""
+    perimeter = committed.union(remaining)
+    greatest_q = greatest_quorum(perimeter, fbas_info['slices_by_node'])
+    if greatest_q == set() or not committed.issubset(greatest_q):
+        return
+    yield frozenset(greatest_q)
+
+    current = greatest_q.difference(committed)
+    while not current == set():
+        # v ← pick from W:
+        node = next_split_node(current, fbas_info['deps_by_node'])
```

Comment:
I think this can further be simplified: Instead of picking the node with the max in-degree we can simply use `pop()` here.

Reply:
pop() doesn't work because it removes the element (and we still need it to be in the `current` set in the next line. 

An alternative seems to be:
```python
# v ← pick from W:
node = next(iter(current))        
```

I'll add a test for the following definition (I guess python allows tests to be skipped by default):
```python
base_quorum_slice_definition = {
    'threshold': 5,
    'validators': [],
    'innerQuorumSets': [
        {'threshold': 2, 'validators': ['blockdaemon-1', 'blockdaemon-2', 'blockdaemon-3'], 'innerQuorumSets': []},
        {'threshold': 2, 'validators': ['coinqvest-de', 'coinqvest-fi', 'coinqvest-hk'], 'innerQuorumSets': []},
        {'threshold': 2, 'validators': ['satoshipay-de-fra', 'satoshipay-sg-sin', 'satoshipay-us-iowa'], 'innerQuorumSets': []},
        {'threshold': 2, 'validators': ['sdf1', 'sdf2', 'sdf3'], 'innerQuorumSets': []},
        {'threshold': 3, 'validators': ['lobstr1', 'lobstr2', 'lobstr3', 'lobstr4', 'lobstr5'], 'innerQuorumSets': []},
        {'threshold': 2, 'validators': ['keybase0', 'keybase1', 'keybase2'], 'innerQuorumSets': []},
    ]
}
```
and see if this decreases the running time from several minutes (I've exited the test after waiting ~5 mins...) to a just few seconds.

Reply:
As per 41d14e4 the above test finishes in a few seconds on my machine 🎉